pkg/compliance/checks/builder.go

```diff
@@ -400,3 +409,117 @@ func (b *builder) baseCheck(ruleID string, kind checkKind, ruleScope string, met
 func newCheckID(ruleID string, kind checkKind) check.ID {
 	return check.ID(fmt.Sprintf("%s:%s", ruleID, kind))
 }
+
+func (b *builder) Reporter() compliance.Reporter {
+	return b.reporter
+}
+
+func (b *builder) DockerClient() env.DockerClient {
+	return b.dockerClient
+}
+
+func (b *builder) AuditClient() env.AuditClient {
+	return b.auditClient
+}
+
+func (b *builder) KubeClient() env.KubeClient {
+	return b.kubeClient
+}
+
+func (b *builder) Hostname() string {
+	return b.hostname
+}
+
+func (b *builder) EtcGroupPath() string {
+	return b.etcGroupPath
+}
+
+func (b *builder) NormalizePath(path string) string {
+	if b.pathMapper == nil {
+		return path
+	}
+	return b.pathMapper(path)
+}
+
+func (b *builder) ResolveValueFrom(valueFrom compliance.ValueFrom) (string, error) {
+	for _, source := range valueFrom {
+		key := source.String()
+		if value, exists := b.valueCache.Get(key); exists {
+			return value.(string), nil
+		}
+
+		value, err := b.getValueFromSource(source)
+		if err != nil {
+			log.Debugf("Failed to fetch %s: %v", key, err)
+			continue
+		}
+
+		b.valueCache.Set(key, value, cache.DefaultExpiration)
+		return value, nil
+
+	}
+	return "", errors.New("failed to resolve")
+}
+
+func (b *builder) getValueFromSource(source compliance.ValueSource) (string, error) {
+	switch {
+	case source.Command != nil:
+		return b.getValueFromCommand(source.Command)
+	case source.Process != nil:
+		return b.getValueFromProcess(source.Process)
+	case source.File != nil:
+		return b.getValueFromFile(source.File)
+	}
+	return "", errors.New("unsupported value source")
+}
+
+func (b *builder) getValueFromCommand(cmd *compliance.ValueFromCommand) (string, error) {
```

Comment:
I believe this duplicates code from the each `Check` class. I believe that, ideally, this should be a static method of the `Command` class, which would be also used by the `Check` itself.
As it does not exist in Go, maybe we can try to put it as a free function in each check file. Knowing that, at some point, we may need to move checks to their own package.

Reply:
Good point, thank you, let me see what can be done to duplicated code.

Reply:
So it is actually using a number of helpers but the logic is somewhat simplified. I am tempted to keep it as is. Once the condition evaluation logic is implemented these will undergo some changes anyway.